### PR TITLE
Api

### PR DESCRIPTION
**Pull request by rega1237**
# The PR for the front-end is [FRONT END REPO](https://github.com/rega1237/hello-react-front-end/pull/1)
This PR includes the following requirements:

# General requirements
- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config). Note for rails api projects without any styling files, delete the stylint configuration in the linter.yml file (line 23 -36) to avoid errors
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
# Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
- Project requirements
 - Create a new Rails API app called 'hello-rails-back-end'.
 - Initialize your project with Git.
 - Make sure that your project has a Postgres database set up.
 - Create a table for storing your messages and create 5 different greetings.
 - Create an API endpoint that selects a random greeting from your table (you will need a controller with an action an rails route).

### CODE REVIEW

File: app/controllers/greetings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class GreetingsController < ApplicationController
+  def index
+    greeting = Greeting.all.sample.message
+    render json: { greeting: }, status: 200
+  end
+end
```
Comment (OLIPLICHE) [inline app/controllers/greetings_controller.rb:6]: # Please Display the random greeting in your Greeting component.
- Both Server is well connected. Make sure that the greetings are displayed. 

  
WhatsApp Image 2022-09-13 at 10 10 42 PM

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @rega1237 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Both servers are running simultaneously without an error
- Linter is green (the last commit have a green check mark (✅)) 👏
- README files (Front and Backend) are descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used  👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/greetings_controller.rb (hunk 1 above)
Comment (rega1237) [reply app/controllers/greetings_controller.rb:6]: Hello @OLIPLICHE ! thanks for the review, but i have a quiestion why you have the const url in the reducer without the '/greet' thats the root where i call to the GET the random message (see my capture from the front end repo)
image

if you open the react server on your browser you gonna see something like this:
image
as you can see in the index i dont have the call to the api is in the click on the link and that sent you to the '/greet' and here is where the recuder make the call action GET to the server
image
image

Comment (OLIPLICHE) [reply app/controllers/greetings_controller.rb:6]: - Hi @rega1237. Nice Job. The Url Link in the reducer is the localhost link I copied when I ran your API in local machine. "Maybe" I omitted to call the root. As I mentioned earlier, your servers are well connected without an error. Great Job for raising this. You can submit it. The next reviewer gonna review it.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @rega1237,

✔️ Good documentation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**